teuthology/task/install/rpm.py

```diff
@@ -148,48 +146,37 @@ def _update_package_list_and_install(ctx, remote, rpm, config):
         _yum_set_check_obsoletes(remote)
 
     if dist_release in ['opensuse', 'sle']:
-        remote.run(
-            args=[
-                'sudo', 'zypper', 'clean', '-a',
-            ])
+        remote.run(args='sudo zypper clean -a')
     else:
-        remote.run(
-            args=[
-                'sudo', 'yum', 'clean', 'all',
-            ])
+        remote.run(args='sudo yum clean all')
 
     ldir = _get_local_dir(config, remote)
 
     if dist_release in ['opensuse', 'sle']:
-        pkg_remove_cmd = 'sudo zypper -n remove --capability {pkg}'
-        pkg_install_cmd = 'sudo zypper -n --no-gpg-checks install --capability --no-recommends {pkg}'
+        remove_cmd = 'sudo zypper -n remove --capability {pkg}'
+        install_cmd = 'sudo zypper -n --no-gpg-checks install --capability --no-recommends {pkg}'
     else:
-        pkg_remove_cmd = 'sudo yum -y remove {pkg}'
-        pkg_install_cmd = 'sudo yum -y install {pkg}'
+        remove_cmd = 'sudo yum -y remove {pkg}'
+        install_cmd = 'sudo yum -y install {pkg}'
 
     for cpack in rpm:
-        pkg = None
         if ldir:
             pkg = "{ldir}/{cpack}".format(
                 ldir=ldir,
                 cpack=cpack,
             )
-            remote.run(
-                args=chain(
-                    ['if', 'test', '-e', run.Raw(pkg), run.Raw(';'), 'then'],
-                    pkg_remove_cmd.format(pkg=pkg).split(), [run.Raw(';')],
-                    pkg_install_cmd.format(pkg=pkg).split(), [run.Raw(';')],
-                    ['fi'])
-            )
-        if pkg is None:
-            remote.run(args=pkg_install_cmd.format(pkg=cpack).split())
+            remote.run(args='''
+              if test -e {pkg} ; then
+                {remove_cmd_pkg} ;
+                {install_cmd_pkg} ;
+              else
+                {install_cmd_cpack} ;
+              fi
+            '''.format(remove_cmd_pkg=remove_cmd.format(pkg=pkg),
+                       install_cmd_pkg=install_cmd.format(pkg=pkg),
+                       install_cmd_cpack=install_cmd.format(pkg=cpack)))
```

Comment:
What about this?
```
remote.run(args='''
    if test -e {ldir}/{pkg} ; then
        {remove_cmd} {ldir}/{pkg}
        {install_cmd} {ldir}/{pkg}
    else
        {install_cmd} {pkg}
    fi
'''.format(ldir=ldir, pkg=cpack,
        install_cmd=install_cmd,
        remove_cmd=remove_cmd))
```

Reply:
@kshtsk i revised the change based on your proposal.